lib/cli-engine.js

```diff
@@ -162,14 +165,33 @@ function processFile(filename, configHelper) {
         config,
         text,
         messages,
-        stats;
+        stats,
+        fileExtension = path.extname(filename),
+        processor;
 
     if (fs.existsSync(filePath)) {
         debug("Linting " + filePath);
         config = configHelper.getConfig(filePath);
         loadPlugins(config.plugins);
         text = fs.readFileSync(path.resolve(filename), "utf8");
-        messages = eslint.verify(text, config, filename);
+
+        for (var plugin in loadedPlugins) {
+            if (loadedPlugins[plugin].processors && Object.keys(loadedPlugins[plugin].processors).indexOf(fileExtension) >= 0) {
+                processor = loadedPlugins[plugin].processors[fileExtension];
+                break;
+            }
+        }
+
+        if (processor) {
+            var parsedBlocks = processor.preprocess(text);
```

Comment:
`Postprocess` is not about merging parts together. It's about adjusting error locations. Even if you stripped out all of the HTML from the file and got one chunk of JavaScript in the `preprocess`, you still have to pad locations of the error messages with the stripped out HTML offsets. Same applies to Coffeescript as well, I think. It doesn't really matter if `preprocess` returned just one item, `postprocess` still needs to clean up.